include/deal.II/particles/particle_accessor.h

```diff
@@ -45,11 +45,44 @@ namespace Particles
   class ParticleAccessor
   {
   public:
+    /**
+     * Data structure to describes the particles in a given cell. This is used
+     * inside an std::list in `particle_container`.
+     */
+    struct ParticlesInCell
+    {
+      /**
+       * Default constructor.
+       */
+      ParticlesInCell() = default;
+
+      /**
+       * Construct from a vector of particles and a cell iterator.
+       */
+      ParticlesInCell(
+        const std::vector<typename PropertyPool<dim, spacedim>::Handle>
+          &particles,
+        const typename Triangulation<dim, spacedim>::active_cell_iterator
+          &cell_iterator)
+        : particles(particles)
+        , cell_iterator(cell_iterator)
+      {}
+
+      /**
+       * A vector of particles on a cell.
+       */
+      std::vector<typename PropertyPool<dim, spacedim>::Handle> particles;
```

Comment:
No, this is the actual storage place of the particle. It would morally belong to `ParticleHandler` where the `particle_container` is actually stored, but moved it into `ParticleAccessor` in order not to touch the order of include files or do some strange forward declaration (`particle_accessor.h` is included by `particle_handler.h`). If you find this confusing, we can move `ParticlesInCell` into a new header file, and make sure to let both files include it. What do you think?